app/controllers/listings_controller.rb

```diff
@@ -0,0 +1,47 @@
+class ListingsController < ApplicationController
+  before_action :call_listing, only: [:show, :create]
+  def index
+    @listings = Listing.all
```

Comment:
For indexing all listings of a user, we will need to use the `current_user` helper. For example we can call `@listings = current_user.listings`

Reply:
Okay. I will fix this... is there anything else I need to be fixing...?

Reply:
Sorry I just added another one for the part `<%= @user.name %>` above. I think `@user` will need to be changed to `current_user` as well.